src/js/map/maps.js

```diff
@@ -127,29 +127,29 @@ export class MapControl extends Component {
      * Handles creating a choropleth when a subindicator is clicked
      * @param  {[type]} data    An object that contains subindictors and obj
      */
-    handleChoropleth(data, method, selectedSubindicator, indicatorTitle, showMapchip, filter) {
+    handleChoropleth(data, method, selectedSubindicator, indicatorTitle, showMapchip, filter, metadata, config) {
         const args = {
             data: data,
+            metadata: metadata,
             selectedSubindicator: selectedSubindicator,
             indicatorTitle: indicatorTitle,
             showMapchip: showMapchip,
-            description: data.metadata.description,
-            filter: filter
+            filter: filter,
+            config: config
         }
 
-        let allSubindicators = args.data.metadata.groups.filter(x => {
-            return x.name === data.metadata.primary_group
+        let allSubindicators = metadata.groups.filter(x => {
+            return x.name === metadata.primary_group
         })[0].subindicators;
 
+        this.displayChoropleth(data, metadata.primary_group, method, selectedSubindicator, allSubindicators, config);
         this.triggerEvent("map.choropleth.loaded", args);
-
-        this.displayChoropleth(data, data.metadata.primary_group, method, selectedSubindicator, allSubindicators);
```

Comment:
Nice change, it doesn't make sense to trigger the loaded event before displaying the choropleth. Wouldn't this have caused a test to fail?

Reply:
the only feature we use loaded event at the moment is while displaying mapchip so it didn't create problems - but you are right it was a possible race condition for future developments. that's why I made this change